lib/thor/line_editor/readline.rb

```diff
@@ -2,8 +2,8 @@ class Thor
   module LineEditor
     class Readline < Basic
       def self.available?
-        begin
-          require "readline"
```

Comment:
`require` by default only happens once. Why do we need this memoization?

Reply:
I had myself convinced that subsequent calls after building the mock would raise a `TypeError` here, but I was incorrect. Will update this PR.